src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java

```diff
@@ -437,6 +443,35 @@ public WindowTownHall(final BuildingTownHall.View townHall)
         registerButton(BUTTON_REMOVE_BLOCK, this::removeBlock);
     }
 
+    /**
+     * Clears and resets all users.
+     */
+    private void updateUsers()
+    {
+        users.clear();
+        users.addAll(townHall.getColony().getPlayers().values());
+        users.sort(Comparator.comparing(Player::getRank, Rank::compareTo));
+    }
+
+    /**
+     * Clears and resets all citizens.
+     */
+    private void updateCitizens()
+    {
+        citizens.clear();
+        citizens.addAll(townHall.getColony().getCitizens().values());
+    }
+
+    /**
+     * Clears and resets all citizens.
+     */
+    private void updateWorkOrders()
+    {
+        workOrders.clear();
+        workOrders.addAll(townHall.getColony().getWorkOrders());
+        workOrders.sort((first, second) -> second.getPriority() > first.getPriority() ? 1 : (second.getPriority() < first.getPriority() ? -1 : 0));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3358)